From fdf4ac05e3b237e407ca11dedb88d550ad08aabb Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Wed, 17 May 2017 09:40:04 -0400
Subject: [PATCH 06/86] arm: dts: am572x: add am572x-evm-jailhouse dts

This commit add the am572x-evm-jailhouse.dts which is extension of the
am57xx-evm-reva3.dts. It allows to demonstrate Jailhouse hypervisor and run
couple of bare-metal as well as TI PDK test application.

1) reserve memory for jailhouse and jailhouse inmate
2) disable timer8 at Linux but left it power-on
3) disable uart9 (with some evm modification uart9 may be used by an
   inmate)
4) skip interrupts 134 and 135 (they will be used by inmates)

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/am572x-evm-jailhouse.dts | 43 ++++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)
 create mode 100644 arch/arm/boot/dts/am572x-evm-jailhouse.dts

diff --git a/arch/arm/boot/dts/am572x-evm-jailhouse.dts b/arch/arm/boot/dts/am572x-evm-jailhouse.dts
new file mode 100644
index 0000000..fba6cba
--- /dev/null
+++ b/arch/arm/boot/dts/am572x-evm-jailhouse.dts
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2017 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "am57xx-evm-reva3.dts"
+
+/ {
+	reserved-memory {
+		jailhouse: jailhouse@ef000000 {
+			reg = <0x0 0xef000000 0x0 0x1000000>;
+			no-map;
+			status = "okay";
+		};
+
+		jh_inmate: jh_inmate@ee000000 {
+			reg = <0x0 0xe0000000 0x0 0xf000000>;
+			no-map;
+			status = "okay";
+		};
+	};
+};
+
+&timer8 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+&uart9 {
+	status = "disabled";
+	ti,no-idle;
+};
+
+/ {
+	ocp {
+		crossbar_mpu: crossbar@4a002a48 {
+			ti,irqs-skip = <10 133 134 135 139 140>;
+		};
+	};
+};
-- 
1.9.1

